storage/aptosdb/src/state_merkle_db.rs

```diff
@@ -773,14 +804,39 @@ impl TreeReader<StateKey> for StateMerkleDb {
 
 impl TreeWriter<StateKey> for StateMerkleDb {
     fn write_node_batch(&self, node_batch: &NodeBatch) -> Result<()> {
-        // TODO(grao): Support sharding here.
         let _timer = OTHER_TIMERS_SECONDS
             .with_label_values(&["tree_writer_write_batch"])
             .start_timer();
-        let batch = SchemaBatch::new();
+        // Get the top level batch and sharded batch from raw NodeBatch
+        let top_level_batches = SchemaBatch::new();
+        let mut jmt_shard_batches: Vec<SchemaBatch> = Vec::with_capacity(NUM_STATE_SHARDS);
+        for _ in 0..NUM_STATE_SHARDS {
+            jmt_shard_batches.push(SchemaBatch::new());
+        }
         node_batch.iter().try_for_each(|(node_key, node)| {
-            batch.put::<JellyfishMerkleNodeSchema>(node_key, node)
+            if let Some(shard_id) = node_key.get_shard_id() {
+                jmt_shard_batches[shard_id as usize]
+                    .put::<JellyfishMerkleNodeSchema>(node_key, node)
+            } else {
+                top_level_batches.put::<JellyfishMerkleNodeSchema>(node_key, node)
+            }
         })?;
-        self.metadata_db().write_schemas(batch)
+
+        TREE_COMMIT_POOL.scope(|s| {
+            let mut state_merkle_batch = jmt_shard_batches.into_iter();
+            for shard_id in 0..NUM_STATE_SHARDS {
+                let state_merkle_batch = state_merkle_batch.next().unwrap();
+                s.spawn(move |_| {
+                    self.state_merkle_db_shards[shard_id]
+                        .write_schemas(state_merkle_batch)
+                        .unwrap_or_else(|_| {
+                            panic!("Failed to commit state merkle shard {shard_id}.")
+                        });
+                });
+            }
+        });
+
+        self.state_merkle_metadata_db
+            .write_schemas(top_level_batches)
```

Comment:
nit: make a `StateMerkleDb::commit_without_progress()` alongside `StateMerkleDb::commit`  to make the difference obvious.